### ISSUE TITLE
Always materialize ARN's when possible

### ISSUE THREAD

**Issue by nick123pig** [cloudquery/cq-provider-aws#123]
I am using cloudquery for IAM policy analysis, and all policies references ARN's in some way. It would be nice to always have the ARN as a materialized column in cloudquery, rather than having to piece it together myself. Much like "tags" is now a default column on most tables, would it be possible to encourage/enforce all tables to have ARN's?

**Comment by roneli** [comment]
Yes that makes a lot of sense, we did add it to a few resources that didn't have ARN by default, we do have a helper function that allows you to build ARN easily. 

Will add ARNs to most resources in the next version

**Comment by roneli** [comment]
WIP #124  adding arns to all resources

**Comment by bbernays** [comment]
Here is a query to find all tables that do not have an ARN field. I have limited it to only tables that have a tag field as well in order to make the first pass more manageable:

```
SELECT DISTINCT TABLE_NAME
FROM INFORMATION_SCHEMA.COLUMNS
WHERE COLUMN_NAME like '%tags%'
	AND DATA_TYPE = 'jsonb'
	AND TABLE_NAME not in
		(SELECT TABLE_NAME
			FROM INFORMATION_SCHEMA.COLUMNS
			WHERE TABLE_SCHEMA = 'public'
				AND COLUMN_NAME like 'arn')
ORDER BY TABLE_NAME ASC
```
 

- [ ] aws_apigateway_api_keys
- [ ] aws_apigateway_client_certificates
- [ ] aws_apigateway_domain_names
- [ ] aws_apigateway_rest_api_stages
- [ ] aws_apigateway_rest_apis
- [ ] aws_apigateway_usage_plans
- [ ] aws_apigateway_vpc_links
- [ ] aws_apigatewayv2_api_stages
- [ ] aws_apigatewayv2_apis
- [ ] aws_apigatewayv2_domain_names
- [ ] aws_apigatewayv2_vpc_links
- [ ] aws_cognito_identity_pools
- [ ] aws_directconnect_connections
- [ ] aws_directconnect_lags
- [ ] aws_directconnect_virtual_interfaces
- [ ] aws_ec2_ebs_snapshots
- [ ] aws_ec2_eips
- [ ] aws_ec2_flow_logs
- [ ] aws_ec2_images
- [ ] aws_ec2_instances
- [ ] aws_ec2_internet_gateways
- [ ] aws_ec2_nat_gateways
- [ ] aws_ec2_network_acls
- [ ] aws_ec2_route_tables
- [ ] aws_ec2_subnets
- [ ] aws_ec2_transit_gateway_attachments
- [ ] aws_ec2_transit_gateway_multicast_domains
- [ ] aws_ec2_transit_gateway_peering_attachments
- [ ] aws_ec2_transit_gateway_route_tables
- [ ] aws_ec2_transit_gateway_vpc_attachments
- [ ] aws_ec2_transit_gateways
- [ ] aws_ec2_vpc_endpoints
- [ ] aws_ec2_vpc_peering_connections
- [ ] aws_ec2_vpcs
- [ ] aws_ec2_vpn_gateways
- [ ] aws_ecs_cluster_container_instances
- [ ] aws_efs_filesystems
- [ ] aws_elbv1_load_balancers
- [ ] aws_guardduty_detectors
- [ ] aws_iam_virtual_mfa_devices
- [ ] aws_rds_clusters
- [ ] aws_redshift_clusters
- [ ] aws_redshift_subnet_groups
- [ ] aws_route53_domains

**Comment by paul-e-allen** [comment]
I second this approach to add ARNs whenever they exist for a resource. In my case, I was specifically surprised that aws_ec2_vpcs didn't contain an ARN field.

**Comment by bbernays** [comment]
@paul-e-allen - we have 2 PRs that have been merged to the main branch, but not released yet that add hundreds of ARN fields to various AWS resources.

I can confirm that we are adding the ARN field to aws_vpc

https://github.com/cloudquery/cq-provider-aws/pull/378
https://github.com/cloudquery/cq-provider-aws/pull/383


If you find any other resources that you feel need ARN's then please let us know!!

**Comment by bbernays** [comment]
As of `v0.10.1` we are only missing the following resources:
```
aws_ec2_ebs_snapshots
aws_ec2_eips
aws_ec2_transit_gateway_attachments
aws_ec2_transit_gateway_multicast_domains
aws_ec2_transit_gateway_peering_attachments
aws_ec2_transit_gateway_route_tables
aws_ec2_transit_gateway_vpc_attachments
aws_ecs_cluster_container_instances
aws_iam_virtual_mfa_devices
aws_route53_domains
```
